### PR TITLE
#68 - Pick project manager from the repository provider.

### PR DESCRIPTION
**Pull request by BogdanAdrian11**
Pull request for #68.
Implemented the provider to be taken into account when choosing a PM for a Repo.
Left puzzle in InMemoryProjectManagers class for adding ProjectManager from each repository and to update the unit test.

### CODE REVIEW

File: self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryProjectManagers.java [hunk 1]
```diff
@@ -35,6 +35,8 @@
  * @author Mihai Andronache (amihaiemil@gmail.com)
  * @version $Id$
  * @since 0.0.1
+ * @todo #68:30min Add ProjectManager from each repository to the pms map.
+ *  Refactor {@link this#pick(String)} to take into consideration the provider.
```
Comment (amihaiemil) [inline self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryProjectManagers.java:39]: @BogdanAdrian11 TODO-ul acesta nu e corect. ProjectManager-ul trebuie sa aiba si el metoda ``String :: provider()``. Poti corecta? :D 

Un Repo e independent si nu are un ProjectManager. Un Project are PM. Practic un Repo debine Project cand este activat, si acest Project are un PM.

File: self-api/src/main/java/com/selfxdsd/api/Repo.java [hunk 2]
```diff
@@ -51,4 +51,10 @@
      * @return Project.
      */
     Project activate();
+
+    /**
+     * Provider name of this repository.
+     * @return Provider.
+     */
+    String provider();
```
Comment (amihaiemil) [inline self-api/src/main/java/com/selfxdsd/api/Repo.java:59]: @BogdanAdrian11 Provider-ul unui Repo se poate lua din owned: ``repo.owner().provider()``... dar cred ca e ok si sa avem metoda asta wrapper.

Comment (BogdanAdrian11) [reply self-api/src/main/java/com/selfxdsd/api/Repo.java:59]: Atunci cred ar fi mai bine ca metoda `String provider()` sa returneze `owner().provider().name()`.

File: self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryProjectManagers.java (hunk 1 above)
Comment (BogdanAdrian11) [reply self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryProjectManagers.java:39]: Sigur, o sa modific.

File: self-api/src/main/java/com/selfxdsd/api/Repo.java (hunk 2 above)
Comment (amihaiemil) [reply self-api/src/main/java/com/selfxdsd/api/Repo.java:59]: @BogdanAdrian11 da da, se poate. E mai elegant :)

**Comment by amihaiemil** [comment]
@BogdanAdrian11 nice, thanks

**Comment by amihaiemil** [comment]
@rultor merge please

**Pull request merged**